File: src/main/java/ru/ezhov/rocket/action/RocketActionUiRepository.java
package ru.ezhov.rocket.action;

import ru.ezhov.rocket.action.api.RocketActionUi;

import java.util.List;
import java.util.Optional;

public interface RocketActionUiRepository {
    List<RocketActionUi> all();

    Optional<RocketActionUi> by(String type);
}

File: src/main/kotlin/ru/ezhov/rocket/action/domain/RocketActionUiRepository.kt
package ru.ezhov.rocket.action.domain

import ru.ezhov.rocket.action.api.RocketActionFactoryUi
import ru.ezhov.rocket.action.api.RocketActionType

interface RocketActionUiRepository {
    fun all(): List<RocketActionFactoryUi>

    fun by(type: RocketActionType): RocketActionFactoryUi?
}